### PR TITLE
feat: setup Southwark with S3 Power Automate integration tokens plus howto doc

### PR DESCRIPTION
**Pull request by jessicamcinchak**
Southwark is ready to replicate Barnet's work on the S3/Power Automate integrations - this sets up 4x API tokens for them to do so. See how-to doc for further explanation.

All encrypted values have been added to `teams_integration` table via prod Hasura console & synce to staging already. AWS S3 `pizza-secrets` has also been updated with staging file api key values.

### CODE REVIEW

**Comment by jessicamcinchak** [comment]
[Setup Southwark to use S3/Power Automate submission option](https://trello.com/c/rhL3PLOe/3120-setup-southwark-to-use-s3-power-automate-submission-option)

File: doc/how-to/how-to-setup-aws-s3-submissions.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Context
+
+When building a submission service, the "Send" component offers an option "Upload to AWS S3" which uploads the application data payload to PlanX's private S3 Bucket - which then in turn allows a council to receive a notification via a Power Automate Webhook and run a script to securely retrieve the application JSON file, download any associated user-uploaded files, generate custom documents, and save the application to their local Microsoft SharePoint environment.
+
+There's full documentation on how this method works in the PlanX Service Specification on Notion here.
+
+Once a council has confirmed they're cleared from their IT dept to use this method (eg allowed to receive external webhook requests) & shared a Power Automate Webhook URL, there's a few steps we need to take:
+
+1. Add the council-provied URL to `team_integrations.power_automate_webhook_url` as plain text via the production Hasura console
```
Comment (jamdelion) [inline doc/how-to/how-to-setup-aws-s3-submissions.md:11]:
```suggestion
1. Add the council-provided URL to `team_integrations.power_automate_webhook_url` as plain text via the production Hasura console
```

File: api.planx.uk/.env.test.example [hunk 2]
```diff
@@ -20,6 +20,7 @@ AWS_SECRET_KEY=👻
 FILE_API_KEY=👻
 FILE_API_KEY_NEXUS=👻
 FILE_API_KEY_BARNET=👻
+FILE_API_KEY_SOUTHWARK=👻
```
Comment (DafyddLlyr) [inline api.planx.uk/.env.test.example:23]: Probably about time we moved these to `team_integrations` as well I think 👍

File: api.planx.uk/modules/auth/middleware.ts [hunk 3]
```diff
@@ -87,7 +87,14 @@ export const useFilePermission: RequestHandler = (req, _res, next): void => {
...
       process.env.FILE_API_KEY_NEXUS!,
     ) ||
-    isEqual(req.headers["api-key"] as string, process.env.FILE_API_KEY_BARNET!);
+    isEqual(
+      req.headers["api-key"] as string,
+      process.env.FILE_API_KEY_BARNET!,
+    ) ||
+    isEqual(
+      req.headers["api-key"] as string,
+      process.env.FILE_API_KEY_SOUTHWARK!,
```
Comment (DafyddLlyr) [inline api.planx.uk/modules/auth/middleware.ts:96]: Ideally we'd also constrain this to Southwark requests, but the current S3 setup doesn't support this.

**Review by DafyddLlyr — APPROVED** [review]

File: doc/how-to/how-to-setup-aws-s3-submissions.md (hunk 1 above)
Comment (jamdelion) [reply doc/how-to/how-to-setup-aws-s3-submissions.md:11]: Actually I did just google 'provied' and apparently it is a word so sorry if that was actually what you intended here!


> definition.com.co
> http://www.definition.com.co › provied
> 1. [ adjective ] established beyond doubt. Synonyms. proven. Examples. "a proven liar" "a Soviet leader of proven shrewdness" ; *. Similar Spelling. proved ...
>

**Review by jamdelion — APPROVED** [review]
Love to see doc improvements 🙌

File: doc/how-to/how-to-setup-aws-s3-submissions.md (hunk 1 above)
Comment (jessicamcinchak) [reply doc/how-to/how-to-setup-aws-s3-submissions.md:11]: An actual typo - thanks for catching !

File: doc/how-to/how-to-setup-aws-s3-submissions.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# How to setup AWS S3 submissions
+
+## Context
+
+When building a submission service, the "Send" component offers an option "Upload to AWS S3" which uploads the application data payload to PlanX's private S3 Bucket - which then in turn allows a council to receive a notification via a Power Automate Webhook and run a script to securely retrieve the application JSON file, download any associated user-uploaded files, generate custom documents, and save the application to their local Microsoft SharePoint environment.
+
+There's full documentation on how this method works in the PlanX Service Specification on Notion here.
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply doc/how-to/how-to-setup-aws-s3-submissions.md]: Yes thanks Notion link added now :link: :heavy_check_mark:

**Pull request merged**